services/user/Homepage/ui/src/apps/contacts/components/contact-details.tsx

```diff
@@ -0,0 +1,266 @@
+import { Avatar, AvatarImage } from "@/components/ui/avatar";
+import { Button } from "@/components/ui/button";
+import {
+    ArrowLeft,
+    Copy,
+    Edit,
+    Mail,
+    MoreVertical,
+    Phone,
+    Trash,
+    Wallet,
+} from "lucide-react";
+import { useAvatar } from "@/hooks/useAvatar";
+import { z } from "zod";
+import { LocalContact } from "../types";
+import { useProfile } from "@/hooks/useProfile";
+import { useDeleteContact } from "../hooks/useDeleteContact";
+import {
+    Dialog,
+    DialogContent,
+    DialogDescription,
+    DialogFooter,
+    DialogHeader,
+    DialogTitle,
+} from "@/components/ui/dialog";
+import { useState } from "react";
+
+import {
+    DropdownMenu,
+    DropdownMenuContent,
+    DropdownMenuItem,
+    DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { ContactForm } from "./contact-form";
+import { useUpdateContact } from "../hooks/useUpdateContact";
+import { formatNames } from "../utils/formatNames";
+
+interface ContactDetailsProps {
+    contact: z.infer<typeof LocalContact> | undefined;
+    onTransferFunds: (contactId: string) => void;
+    onChainMailUser: (contactId: string) => void;
+    onBack?: () => void;
+}
+
+export function ContactDetails({
+    contact,
+    onTransferFunds,
+    onChainMailUser,
+    onBack,
+}: ContactDetailsProps) {
+    if (!contact) {
+        return (
+            <div className="flex h-full items-center justify-center">
+                Select a contact to view details
+            </div>
+        );
+    }
+
+    const { data: profile } = useProfile(contact.account);
+
+    const avatarSrc = useAvatar(contact.account);
+
+    const [primaryName, secondaryName] = formatNames(
+        contact.nickname,
+        profile?.profile?.displayName,
+        contact.account,
+    );
+
+    console.log({
+        primaryName,
+        secondaryName,
+        nickname: contact.nickname,
+        displayName: profile?.profile?.displayName,
+        account: contact.account,
+    });
+
+    const { mutate: deleteContact } = useDeleteContact();
+    const [showModal, setShowModal] = useState(false);
+    const [isEditing, setIsEditing] = useState(false);
+
+    const { mutateAsync: updateContact } = useUpdateContact();
+
+    return (
+        <div>
+            {onBack && (
+                <div className="pl-4">
+                    <Button onClick={onBack} variant="outline" size="icon">
+                        <ArrowLeft />
+                    </Button>
+                </div>
+            )}
```

Comment:
I recommend reusing the same detail header we're using in chainmail's `message-detail.tsx`. It nicely aligns everything and would lend to a more consistent look and feel between the apps.

Reply:
Adopted

Reply:
Cool. I see it now. We do have an alignment issue though:
<img width="758" alt="image" src="https://github.com/user-attachments/assets/8672ebb0-7fa4-49cf-824c-aca881f880e9" />

I think the button in your version on the right hand side might be larger than the ones we use in Chain mail (for example, on a draft email).

Reply:
Fixed alignment issue, I can't find a size difference though